### ISSUE TITLE
VerifyTypes emits error for unannotated `self` parameter in property

### ISSUE THREAD

**Issue by davfsa** [microsoft/pyright#3262]
### Describe the bug
Type of `self` argument not automatically discovered for properties

### Expected behavior
No errors raised, since self type should be known by pyright.

### Steps to reproduce
`Project Structure`

![image](https://user-images.githubusercontent.com/29100934/160414695-73aec2e3-7e4a-4cdf-9d9f-e2dba3f842b0.png)

`test_bug.py`
```python
class Foo:
    # All good :)
    def bar(self) -> None:
        return

    # Error here :(
    @property
    def baz(self) -> None:
        return

```

`setup.py`
```py
import setuptools

setuptools.setup(
    name="testing",
    version="1.0.0",
    packages=setuptools.find_namespace_packages(include=["testing*"]),
    zip_safe=False,
    include_package_data=True,
)
```

`MANIFEST.in`
```
graft testing
include setup.py
global-exclude *.py[cod]
```

### Terminal output
```
$ python -m pyright --version
pyright 1.1.233

$ python --version
Python 3.9.11

$ python -m pip install .
<redacted for simplicity sake>

$ python -m pyright --verifytypes testing
Package name: "testing"
Package directory: "/home/davfsa/coding/hikari/.venv/lib/python3.9/site-packages/testing"
Path of py.typed file: "/home/davfsa/coding/hikari/.venv/lib/python3.9/site-packages/testing/py.typed"

Public modules: 2
   testing
   testing.test_bug

Symbols used in public interface:
testing.test_bug.Foo.baz
  /home/davfsa/coding/hikari/.venv/lib/python3.9/site-packages/testing/test_bug.py:error: Type annotation for parameter "self" is missing

Symbols exported by "testing": 3
  With known type: 1
  With ambiguous type: 0
  With unknown type: 2
  Functions without docstring: 2
  Functions without default param: 0
  Classes without docstring: 1

Other symbols referenced but not exported by "testing": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0

Type completeness score: 33.3%

Completed in 0.355sec
```
*Note that we started seeing the error since 1.1.232*

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This was caused by a confluence of two different big fixes in the core type checker, and they had unexpected results in the "verifytypes" logic. The problem will be fixed in the next release.

**Comment by davfsa** [comment]
Thanks a lot for taking a look at it!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.234, which I just published.
